core/src/io/anuke/mindustry/ui/dialogs/CustomRulesDialog.java

```diff
@@ -1,67 +1,128 @@
+        f_number("$rules.respawntime", f -> rules.respawnTime = f * 60f, () -> rules.respawnTime / 60f, ()->true);
+        title("$rules.title.resourcesbuilding", Gamemode.attack, Gamemode.pvp, Gamemode.survival, Gamemode.sandbox);
+        check("$rules.infiniteresources", b -> rules.infiniteResources = b, () -> rules.infiniteResources, ()->true, Gamemode.attack, Gamemode.pvp, Gamemode.survival, Gamemode.sandbox);
+        f_number("$rules.buildcostmultiplier", f -> rules.buildCostMultiplier = f, () -> rules.buildCostMultiplier, ()->!rules.infiniteResources, Gamemode.attack, Gamemode.pvp, Gamemode.survival, Gamemode.sandbox);
+        f_number("$rules.buildspeedmultiplier", f -> rules.buildSpeedMultiplier = f, () -> rules.buildSpeedMultiplier, ()->true, Gamemode.attack, Gamemode.pvp, Gamemode.survival, Gamemode.sandbox);
+        title("$rules.title.player");
+        f_number("$rules.playerdamagemultiplier", f -> rules.playerDamageMultiplier = f, () -> rules.playerDamageMultiplier, ()->true);
+        f_number("$rules.playerhealthmultiplier", f -> rules.playerHealthMultiplier = f, () -> rules.playerHealthMultiplier, ()-> true);
+        title("$rules.title.unit");
+        check("$rules.unitdrops", b -> rules.unitDrops = b, () -> rules.unitDrops, ()->true);
+        f_number("$rules.unitbuildspeedmultiplier", f -> rules.unitBuildSpeedMultiplier = f, () -> rules.unitBuildSpeedMultiplier, ()->true);
+        f_number("$rules.unithealthmultiplier", f -> rules.unitHealthMultiplier = f, () -> rules.unitHealthMultiplier, ()->true);
+        f_number("$rules.unitdamagemultiplier", f -> rules.unitDamageMultiplier = f, () -> rules.unitDamageMultiplier, ()->true);
+        title("$rules.title.enemy", Gamemode.attack, Gamemode.pvp);
+        f_number("$rules.enemycorebuildradius", f -> rules.enemyCoreBuildRadius = f * tilesize, () -> Math.min(rules.enemyCoreBuildRadius / tilesize, 200), ()->true, Gamemode.attack, Gamemode.pvp);
     }
 
-    void number(String text, FloatConsumer cons, FloatProvider prov){
+    void f_number(String text, FloatConsumer cons, FloatProvider prov, BooleanProvider condition){
         main.table(t -> {
             t.left();
-            t.add(text).left().padRight(5);
+            t.add(text).left().padRight(5)
+                    .update(a->a.setColor(condition.get() ? Color.WHITE : Color.GRAY));
             Platform.instance.addDialog(t.addField(prov.get() + "", s -> cons.accept(Strings.parseFloat(s)))
-            .valid(Strings::canParsePositiveFloat).width(120f).left().get());
-        });
+                    .padRight(100f)
+                    .update(a -> a.setDisabled(!condition.get()))
+                    .valid(Strings::canParsePositiveFloat).width(120f) .left().get());
+        }).padTop(0);
+        main.row();
+    }
+
+    void f_number(String text, FloatConsumer cons, FloatProvider prov, BooleanProvider condition, Gamemode... gamemodes){
```

Comment:
One last thing: In Java conventions, method names shouldn't have underscores. Instead of `f_number` and `i_number`, use `number` (for floats, since that's most common) and `integer`.